### PR TITLE
Add Variables subcommands for listing and deleting variables

### PR DESCRIPTION
**Pull request by kate-goldenring**
Adds `list` and `delete` subcommands to the `variables` CLI

Also, hides the `variables` subcommand and `--variable` deploy flag. Can unhide when we release the feature

### CODE REVIEW

File: src/commands/variables.rs [hunk 1]
```diff
@@ -73,6 +98,30 @@ impl VariablesCommand {
...
+            Self::List(cmd) => {
+                let info =
+                    AppManagmentInfo::new(cmd.common.deployment_env_id.as_deref(), &cmd.common.app)
+                        .await?;
+                let vars = get_variables(&info.client, info.app_id).await?;
+                let var_names = vars
+                    .iter()
+                    .map(|var| from_str(var))
+                    .collect::<Result<Vec<Variable>, _>>()
+                    .context("could not parse variable")?;
```
Comment (fibonacci1729) [inline src/commands/variables.rs:116]: For consistency with other contexts here:

```suggestion
                    .context("Could not parse variable")?;
```

**Review by fibonacci1729 — APPROVED** [review]

File: src/commands/variables.rs [hunk 2]
```diff
@@ -73,6 +98,30 @@ impl VariablesCommand {
...
+                if !var_names.is_empty() {
+                    println!();
+                    println!("App {} has the following variables:", cmd.common.app)
+                }
+                for v in var_names {
```
Comment (kate-goldenring) [inline src/commands/variables.rs:121]: @itowlson I think i could use help on making this a prettier experience than tabbed prints. Or maybe this is sufficient

Comment (itowlson) [reply src/commands/variables.rs:121]: @kate-goldenring For multi-value data we've used tables (e.g. `spin templates list`), but there doesn't seem much point when we're just listing names (cf. `spin plugins list`).

That said, I'm not sure there's value in the header and tabs.  How would it feel to just have:

```
$ spin cloud variables list
alpha
bravo
charlie
```

If we changed it to show extra info like the value or if the variable was secret or was modified from its default value or whatever, then `comfy-table` has been our friend.

Comment (kate-goldenring) [reply src/commands/variables.rs:117]: Sounds good to me. Simple and clean. Fixed it

File: src/commands/variables.rs [hunk 3]
```diff
@@ -73,6 +98,26 @@ impl VariablesCommand {
...
+                    AppManagmentInfo::new(cmd.common.deployment_env_id.as_deref(), &cmd.common.app)
+                        .await?;
+                let vars = get_variables(&info.client, info.app_id).await?;
+                let var_names = vars
+                    .iter()
+                    .map(|var| from_str(var))
+                    .collect::<Result<Vec<Variable>, _>>()
+                    .context("could not parse variable")?;
+                for v in var_names {
+                    println!("{}", v.key);
```
Comment (radu-matei) [inline src/commands/variables.rs:118]: Not blocking this PR, but do we want to format a table head? For example, the output for the templates list command:
```
+------------------------------------------------------------------------+
| Name                Description                                        |
+========================================================================+

```

**Review by radu-matei — APPROVED** [review]

**Pull request merged**